doc/src/tutorial/codegen_explainatory.rst

```diff
@@ -0,0 +1,458 @@
+        these files are returned. [default: False]
+
+    header : bool, optional
+        When True, a header is written on top of each source file.
+        [default: True]
+
+    empty : bool, optional
+        When True, empty lines are used to structure the code.
+        [default: True]
+
+    argument_sequence : iterable, optional
+        Sequence of arguments for the routine in a preferred order.  A
+        CodeGenError is raised if required arguments are missing.
+        Redundant arguments are used without warning.  If omitted,
+        arguments will be ordered alphabetically, but with all input
+        aguments first, and then output or in-out arguments.
+
+    global_vars : iterable, optional
+        Sequence of global variables used by the routine.  Variables
+        listed here will not show up as function arguments.
+
+Here is an example::
+
+    >>> [(f_name, f_code), header]=codegen(
+    ...     ("volume",length*breadth*height), "F95", header=True, empty=False,
+    ...     argument_sequence=(breadth, length), global_vars=(height,))
+    >>> print(f_code)
+    !******************************************************************************
+    !*                    Code generated with sympy 0.7.7.dev                     *
+    !*                                                                            *
+    !*              See http://www.sympy.org/ for more information.               *
+    !*                                                                            *
+    !*                       This file is part of 'project'                       *
+    !******************************************************************************
+    REAL*8 function volume(breadth, length)
+    implicit none
+    REAL*8, intent(in) :: breadth
+    REAL*8, intent(in) :: length
+    volume = breadth*height*length
+    end function
+
+
+``make_routine`` creates a ``Routine`` object, which represents an evaluation
+routine for a set of expressions. This is only good for internal use by the CodeGen 
+objects, as an intermediate representation from SymPy expression to generated code. 
+It is not recommended to make a ``Routine`` object yourself. You should instead use 
+``make_routine`` method. ``make_routine`` in turn calls the ``routine`` method of 
+the CodeGen object depending upon the language of choice. This creates the internal 
+objects representing assignments and so on, and creates the ``Routine`` class with them.
+
+The various codegen objects such as ``Routine`` and ``Variable`` aren't SymPy 
+objects (they don't subclass from Basic).
+
+For example::
+
+    >>> from sympy.utilities.codegen import make_routine
+    >>> from sympy.physics.hydrogen import R_nl
+    >>> from sympy import *
+    >>> init_printing(use_latex='mathjax')
+    >>> x, y=symbols('x y')
+    >>> expr=R_nl(3,y,x,6)
+    >>> r=make_routine('my_routine',expr)
+    >>> [arg.result_var for arg in r.results]
+    [result₅₁₄₂₃₄₁₆₈₁₃₉₇₇₁₉₄₂₈]
```

Comment:
I can't help it. I tried init_printing without any arguments as you suggested. I still get a similar result. Please help me with this.